controllers/send-widget.controller.es6

```diff
@@ -359,9 +364,23 @@ export default class SendWidgetController {
           .addMemo(memo)
           .build();
 
-        transaction.sign(Keypair.fromSeed(this.session.getSecret()));
-
-        return this.Server.submitTransaction(transaction);
+        if (this.useLedger) {
+          let ledgerApi = new StellarLedger.Api(new StellarLedger.comm(120));
+          return ledgerApi.signTx_async(this.bip32Path, transaction).then(result => {
+            let signature = result['signature'];
+            let keyPair = Keypair.fromAccountId(this.session.address);
+            let hint = keyPair.signatureHint();
+            let decorated = new xdr.DecoratedSignature({hint, signature});
+            transaction.signatures.push(decorated);
+            this.Server.submitTransaction(transaction);
```

Comment:
Should be:
```js
return this.Server.submitTransaction(transaction);
```
Right now, it can lose context if promise in `submitTransaction` is rejected.

Reply:
Good one. I'm not so up to speed with javascript promises.